contracts/nft-loan/src/execute.rs

```diff
@@ -44,10 +44,17 @@ pub fn deposit_collaterals(
     let borrower = info.sender;
 
     // ensure atleas one asset has been provided
+    // additional check: check if the person is really owner + if approved to be handled by contract
+    // need to check if the loan > approval because this could lead to problems
+    // cw721_base::msg::QueryMsg::Approval { token_id: "".to_string(), spender: "".to_string(), include_expired: None, }
+    // cw721_base::msg::QueryMsg::OwnerOf { token_id: "".to_string(), include_expired: None }
+    // same goes for sg721_base
     if tokens.is_empty() {
         return Err(ContractError::NoAssets {});
     }
 
+    // could check that info.funds.len() == 1 because the below does not cover 0
```

Comment:
this check you are describing happens on line 69, where if the fee is less than the expected `deposit_fee_amount`, we return a ContractError. 

Is this function what you aim to enforce here?

Reply:
Works for me, forget about my comment